documents/coredocument.go

```diff
@@ -304,38 +343,22 @@ func (cd *CoreDocument) DocumentRootTree() (tree *proofs.DocumentTree, err error
 		return nil, err
 	}
 
-	// For every signature we create a LeafNode
-	sigProperty := NewLeafProperty(SignaturesField, compactProperties(SignaturesField))
-	sigLeafList := make([]proofs.LeafNode, len(cd.Document.Signatures)+1)
-	sigLengthNode := proofs.LeafNode{
-		Property: sigProperty.LengthProp(proofs.DefaultSaltsLengthSuffix),
-		Salt:     make([]byte, idSize),
-		Value:    []byte(fmt.Sprintf("%d", len(cd.Document.Signatures))),
-	}
-	h := sha256.New()
-	err = sigLengthNode.HashNode(h, true)
+	// Next are leaves from the signature data tree
+	signatureTree, err := cd.getSignatureDataTree()
 	if err != nil {
 		return nil, err
 	}
 
-	sigLeafList[0] = sigLengthNode
-	for i, sig := range cd.Document.Signatures {
-		payload := sha256.Sum256(append(sig.EntityId, append(sig.PublicKey, sig.Signature...)...))
-		leaf := proofs.LeafNode{
-			Hash:     payload[:],
-			Hashed:   true,
-			Property: sigProperty.SliceElemProp(proofs.FieldNumForSliceLength(i)),
-		}
-
-		err = leaf.HashNode(h, true)
-		if err != nil {
-			return nil, err
-		}
-
-		sigLeafList[i+1] = leaf
+	properties := signatureTree.PropertyOrder()
+	signatureTreeLeaves := make([]proofs.LeafNode, len(properties))
+	for _, prop := range properties {
+		i, leaf := signatureTree.GetLeafByProperty(prop.ReadableName())
+		// Need to set leaf hashed otherwise it will be rehashed by tree.Generate() below
+		leaf.Hashed = true
+		signatureTreeLeaves[i] = *leaf
 	}
 
-	err = tree.AddLeaves(sigLeafList)
+	err = tree.AddLeaves(signatureTreeLeaves)
```

Comment:
I thought we would only have to add the roothash of the tree to docRootTree just like signingRoot,
why are we adding all the leaves to the tree?
@lucasvo is it not what you meant in our last discussion?

Reply:
So we can create a proof of the fields directly using the documentroot tree and not having to load a different tree for it.

Reply:
Yes, you're definitely right @vimukthi-git it should just be the root hash of `getSignatureDataTree`

Reply:
https://images.zenhubusercontent.com/56946663638e718c995ffc5e/5c974b8b-d7b7-4111-8713-4adf9edd8cbc

Reply:
@mikiquantum we definitely wanted to achieve the opposite. The goal was not add extra levels to the root tree because performance wise it doesn't make sense.

Reply:
So that means we need to iterate over at max 3 trees to find fields to create proofs.